aspnetcore/blazor/components/index.md

```diff
@@ -268,17 +268,141 @@ In the following example from the sample app, the `ParentComponent` sets the val
 
 [!code-razor[](index/samples_snapshot/ParentComponent.razor?highlight=5-6)]
 
-By convention, an attribute value that consists of C# code is assigned to a parameter using [Razor's reserved `@` symbol](xref:mvc/views/razor#razor-syntax):
+Assign C# fields, properties, and methods to component parameters as HTML attribute values using [Razor's reserved `@` symbol](xref:mvc/views/razor#razor-syntax):
 
-* Parent field or property: `Title="@{FIELD OR PROPERTY}`, where the placeholder `{FIELD OR PROPERTY}` is a C# field or property of the parent component.
-* Result of a method: `Title="@{METHOD}"`, where the placeholder `{METHOD}` is a C# method of the parent component.
-* [Implicit or explicit expression](xref:mvc/views/razor#implicit-razor-expressions): `Title="@({EXPRESSION})"`, where the placeholder `{EXPRESSION}` is a C# expression.
+* To assign a parent component's field or property value to a child component's parameter, prefix the field or property name with the `@` symbol. To assign a C# field named `title` to the `Title` parameter, use the syntax `Title="@title"`, as seen in the following parent component:
+
+  ```razor
+  <ChildComponent Title="@title">
+      Title from field.
+  </ChildComponent>
+  
+  @code {
+      private string title = "Panel Title from Parent";
+  }
+  ```
+
+* To assign the result of a parent component's method to a child component's parameter, prefix the method name with the `@` symbol. To assign the result of a C# method named `GetTitle` to the `Title` parameter, use the syntax `Title="@GetTitle"`, as seen in the following parent component:
+
+  ```razor
+  <ChildComponent Title="@GetTitle()">
+      Title from method.
+  </ChildComponent>
+  
+  @code {
+      private string GetTitle()
+      {
+          return "Panel Title from Parent";
+      }
+  }
+  ```
+  
+* To assign the result of an [implicit C# expression](xref:mvc/views/razor#implicit-razor-expressions) in the parent component to a child component's parameter, prefix the implicit expression with an `@` symbol. To assign the current local date in long format with <xref:System.DateTime.ToLongDateString%2A> in the parent component, assign `@DateTime.Now.ToLongDateString()` to the child's parameter. To assign a `person` object's `Name` property, assign `@person.Name`. The following example demonstrates these concepts:
+
+  ```razor
+  <ChildComponent Title="@DateTime.Now.ToLongDateString()">
+      Title from implicit Razor expression.
+  </ChildComponent>
+  
+  <ChildComponent Title="@person.Name">
+      Title from implicit Razor expression.
+  </ChildComponent>
+  ```
```

Comment:
The above is all absolutely fine and good now, thanks! I do still wonder why we're splitting it into these three cases, since there's nothing in the product code that distinguishes them. As far as the Razor language is concerned, all three of these are simply "using a C# expression", and we don't care whether the expression is a field/property/method-call/complex-expression.

You could even combine all three into a single example, similar to:

----

To assign the result of a C# expression to a child component parameter, use the `@` symbol. As you see in the following example, the expression can be a field, property, method call, or other complex evaluation.

```razor
<ChildComponent Visible="@IsVisible" Title="@ComputeTitle()"  ExpiryDate="@DateTime.Now.AddDays(7)" />
```

Reply:
I'm only concerned on the **_gotcha_** 😈 scenarios that work in a Razor page but not a Razor component parameter assignment. IMO, we *might* want to leave those two split out and combine the rest.

I'll work it again Tuesday morning and ping u back.

Reply:
@SteveSandersonMS ... I'm not so sure about making this change. Here's the deal ... readers are cutting and pasting our code from these topics directly into their own test app to play with the scenarios. Your example changes to combine (and including your prompt to not `ToString` the DT) break that educational experience because they can't use the `ChildComponent` component any longer to perform the testing/hacking/playing with our code. I don't think we should necessarily expand (and complicate) that child component. I'll work with it a bit more and see what I can come up with. I'll at least try combining the simple use cases into a single bullet and example.